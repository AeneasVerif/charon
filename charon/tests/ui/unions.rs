union Foo {
    one: u64,
    two: [u32; 2],
}

fn use_union() {
    let mut one = Foo { one: 42 };
    unsafe {
        one.one = 43;
    }
    let _two = unsafe { one.two };
}
