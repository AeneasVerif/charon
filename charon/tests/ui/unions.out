# Final LLBC before serialization:

// Full name: test_crate::Foo
union Foo {
  one: u64,
  two: Array<u32, 2 : usize>,
}

// Full name: test_crate::use_union
fn use_union()
{
    let @0: (); // return
    let one@1: Foo; // local
    let @2: (); // anonymous local
    let _two@3: Array<u32, 2 : usize>; // local

    storage_live(one@1)
    one@1 := Foo { one: const (42 : u64) }
    storage_live(@2)
    (one@1).one := const (43 : u64)
    storage_dead(@2)
    storage_live(_two@3)
    _two@3 := copy ((one@1).two)
    @0 := ()
    storage_dead(_two@3)
    storage_dead(one@1)
    @0 := ()
    return
}



